extensions/amp-auto-ads/0.1/test/test-ad-strategy.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {AdStrategy} from '../ad-strategy';
+import {PlacementState, getPlacementsFromConfigObj} from '../placement';
+import * as sinon from 'sinon';
+
+describe('amp-auto-ads', () => {
```

Comment:
you might want to use describes.fakeWin
```
describes.fakeWin('xxx', {}, env => {
  
});
```
See [here](https://github.com/ampproject/amphtml/blob/master/test/functional/3p/test-iframe-messaging-client.js) as an example.

This runs your tests in a standalone iframe, so it will not interfere with other tests.
It has does those routine setup, such as sandbox for you.